app/services/course_search_service.rb

```diff
@@ -26,7 +30,12 @@ def call
     # The 'where' scope will remove duplicates
     # An outer query is required in the event the provider name is present.
     # This prevents 'PG::InvalidColumnReference: ERROR: for SELECT DISTINCT, ORDER BY expressions must appear in select list'
-    outer_scope = Course.where(id: scope.select(:id))
+    outer_scope = Course.includes(
+      :enrichments,
+      :subjects,
```

Comment:
```suggestion
      subjects: [:financial_incentive],
```

probaby this on as well

Reply:
Good spot. I didn't see N+1 from that as I didn't select the salary filter when I was testing 🤦 I've updated 👍